network.js

```diff
@@ -42,8 +46,15 @@ export default function trainNeuralNet(input, output, nonlin = sigmoid) {
 
     // add our normalized error to the hidden layer
     synapse = Matrix.add(synapse, layer0.transpose().mmul(layer1Delta))
```

Comment:
when adding the gradient to the hidden layer weights, this too should be multiplied by the learning rate.